interfaces/backends/backends.go

```diff
@@ -31,21 +31,33 @@ import (
 	"github.com/snapcore/snapd/interfaces/udev"
 	"github.com/snapcore/snapd/logger"
 	apparmor_sandbox "github.com/snapcore/snapd/sandbox/apparmor"
+	systemd_tools "github.com/snapcore/snapd/systemd"
 )
 
 // All returns a set of all available security backends.
 func All() []interfaces.SecurityBackend {
+	// Backends have non-obvious ordering constraints. The order in which they
+	// are registered below is significant.  Please refrain from re-ordering
+	// them when re-factoring this code. Because this list changes rarely it is
+	// most likely that interactions are implicit and not well-known. Having
+	// said that we know of one specific constraint that is documented:
+	//
+	// Because of how the GPIO interface is implemented the systemd backend
+	// must be earlier in the sequence than the apparmor backend.
```

Comment:
```suggestion
	// Backends have non-obvious ordering constraints. The order in which they
	// are registered below is significant. Please refrain from reordering
	// them when refactoring this code. Because this list changes rarely it is
	// most likely that interactions are implicit and not well-known. Having
	// said that we know of one specific constraint that is documented:
	//
	// Because of how the GPIO interface is implemented the systemd backend
	// must be earlier in the sequence than the apparmor backend.
```

Reply:
Done